ssg/products.py

```diff
@@ -40,6 +42,36 @@ def parse_name(product):
     return _product, _product_version
 
 
+def get_all(ssg_root):
+    """
+    Analyzes all products in the SSG root and sorts them into two categories:
+    those which use linux_os and those which use their own directory. Returns
+    a tuple of sets, (linux_products, other_products).
```

Comment:
I find a tuple quite restrictive, what about returning a name tuple or dictionary? Then, if the need arises, one can introduce `all`, `browsers` or `java` sets :-)